compositor_common/src/scene/builtin_transformations.rs

```diff
@@ -153,31 +160,26 @@ impl BuiltinSpec {
                 }
                 Ok(())
             }
-            BuiltinSpec::TiledLayout { .. } => {
-                if node_spec.input_pads.is_empty() {
-                    return Err(BuiltinSpecValidationError::TiledLayoutNoInputs);
-                }
-
-                if node_spec.input_pads.len() > 16 {
-                    return Err(BuiltinSpecValidationError::TiledLayoutTooManyInputs);
-                }
-
-                Ok(())
-            }
-            BuiltinSpec::MirrorImage { .. } => {
-                if node_spec.input_pads.len() != 1 {
-                    return Err(BuiltinSpecValidationError::MirrorImageExactlyOneInput);
-                }
-
-                Ok(())
-            }
-            BuiltinSpec::CornersRounding { .. } => {
-                if node_spec.input_pads.len() != 1 {
-                    return Err(BuiltinSpecValidationError::CornerRadiusExactlyOneInput);
-                }
+            BuiltinSpec::TiledLayout { .. }
+            | BuiltinSpec::MirrorImage { .. }
+            | BuiltinSpec::CornersRounding { .. }
+            | BuiltinSpec::TransformToResolution { .. } => Ok(()),
+        }
+    }
 
-                Ok(())
-            }
+    pub fn inputs_constrains(&self) -> InputsCountConstraint {
```

Comment:
```suggestion
    pub fn constrains(&self) -> InputsCountConstraint {
```

Reply:
It should return some wrapper type not input specifically, plus single node can have more than one constraint e.g. `Vec<NodeConstraint>`